server 3.4.0 release (#6446)
circleci/circleci-docs: jekyll/_cci2/server-3-install-build-services.adoc

  * You may also create a nomad gcp service account manually. The service account will need the role `compute.admin`. It will also need the role `iam.workloadIdentityUser` if using link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[Workload identities]
. Set Nomad Autoscaler to `enabled`
. Set Maximum Node Count*
. Set Minimum Node Count*
. Select cloud provider: `Google Cloud Platform`
. Add your Project ID
. Add Managed Instance Group Name
. Instance group type: link:https://cloud.google.com/compute/docs/instance-groups/#types_of_managed_instance_groups[Zonal or Regional].

> . You can choose one of the following:
> .. JSON of GCP service account for Nomad Autoscaler
> .. Or, the Nomad Autoscaler Service Account Email Address if using link:https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[Workload Identities]. Steps to enable Workload Identities on GCP cluster are link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[here].
> .. Enable workload identity for `nomad-autoscaler` (kubernetes) service account
> ```shell
> gcloud iam service-accounts add-iam-policy-binding <YOUR_SERVICE_ACCOUNT_EMAIL> \
>     --role roles/iam.workloadIdentityUser \
>     --member "serviceAccount:<GCP_PROJECT_ID>.svc.id.goog[circleci-server/nomad-autoscaler]"
> ```

Author: ```suggestion
. You can choose one of the following:
.. JSON of GCP service account for Nomad Autoscaler
.. Nomad Autoscaler Service Account email address if using link:https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[Workload Identities]. Steps to enable Workload Identities on a GCP cluster are link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[here].
.. Enable workload identity for `nomad-autoscaler` (kubernetes) service account by running the following:
+
```shell
gcloud iam service-accounts add-iam-policy-binding <YOUR_SERVICE_ACCOUNT_EMAIL> \
    --role roles/iam.workloadIdentityUser \
    --member "serviceAccount:<GCP_PROJECT_ID>.svc.id.goog[circleci-server/nomad-autoscaler]"
```
```

Author: @circleci/server Is this a typo: `svc/id/goog` should be `google` I guess?

===== GCP
. Create a service account for Nomad Autoscaler

>   * Our link:https://github.com/CircleCI-Public/server-terraform/tree/main/nomad-gcp[nomad module] creates a service acount and outputs a file with the keys if you set the variables `nomad_auto_scaler = true` and `enable_workload_identity = false`. You may reference the examples in the link for more details. If you've already created the clients, you can simply update the variable and run `terraform apply`. The created user's key will be available in a file named `nomad-as-key.json`. If you are using GKE link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[Workload Identities], set the variables `nomad_auto_scaler = true` and `enable_workload_identity = true`.

Reviewer: ```suggestion
  * Our link:https://github.com/CircleCI-Public/server-terraform/tree/main/nomad-gcp[nomad module] creates a service acount and outputs a file with the keys if you set the variables `nomad_auto_scaler = true` and `enable_workload_identity = false`. You may reference the examples in the link for more details. If you have already created the clients, simply update the variable and run `terraform apply`. The created user's key will be available in a file named `nomad-as-key.json`. If you are using GKE link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[Workload Identities], set the variables `nomad_auto_scaler = true` and `enable_workload_identity = true`.
```

* *Nomad Server Certificate Authority (CA) Certificate (required)* -
Provided in the output from `terraform apply`

* *Build Agent Image* - 

> Consult customer support to set up

Author: ```suggestion
If you want to use a custom Docker registry to supply the CircleCI Build Agent, contact customer support for assistance.
```

Author: @circleci/server
